wmfdata/spark.py

```diff
@@ -326,10 +315,7 @@ def run(commands, format="pandas", session_type="yarn-regular", extra_settings={
             overall_result = cmd_result
     
     if overall_result:
-        if format == "pandas":
-            overall_result = overall_result.toPandas()
-        elif format == "raw":
-            overall_result = overall_result.collect()
+        overall_result = overall_result.toPandas()
```

Comment:
So the `run` functions here and in the MariaDB, Hive, and Presto modules are designed to be simple "write SQL, get a Pandas DataFrame" interfaces. With any of those you could pull down too much data and get an OOM (just as you could try loading a huge Parquet file into Pandas and get an OOM). So in that sense this function is just behaving as it has always done. I will mention it in the docs, though.

The point about returning a Spark dataframe is interesting! I don't think we need to do it here, since the function has never supported it before and we have enough stuff to take care of in this release 😊 But in the future we could potentially add a "spark" format that does what you suggest. I'm not sure how valuable it would be, since if you are interacting with the PySpark API directly you know how to use its `sql` method. But I'm open to the discussion.

> Also, if we keep just toPandas(), should we define the return type?
Sure, seems like a good idea!

Reply:
👍

Reply:
FYI, I'm going to update the docs and add the type hints in #36, since it seems like it'll be more on topic there.